File: C#/739. Daily Temperatures.cs
public class Solution {
    public int[] DailyTemperatures(int[] temperatures) {
        int n = temperatures.Length;
        int[] arr = new int[n];
        Stack<int> st = new Stack<int>();
        for(int i = 0; i < n; i++){
            while(st.Count > 0){
                if(temperatures[i] > temperatures[st.Peek()]){
                    arr[st.Peek()] = i - st.Peek();
                    st.Pop();
                }
                else{
                    break;
                }
            }
            st.Push(i);
        }
        return arr;
    }
}

File: Java/739. Daily Temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        int[] arr = new int[n];
        Stack<Integer> st = new Stack<Integer>();
        for(int i = 0; i < n; i++){
            while(st.size() > 0){
                if(temperatures[i] > temperatures[st.peek()]){
                    arr[st.peek()] = i - st.peek();
                    st.pop();
                }
                else{
                    break;
                }
            }
            st.push(i);
        }
        return arr;
    }
}
